drivers/soundwire/bus.c

```diff
@@ -1845,6 +1845,7 @@ void sdw_clear_slave_status(struct sdw_bus *bus, u32 request)
 		if (slave->status != SDW_SLAVE_UNATTACHED) {
```

Comment:
The commit message is a bit hard to read. How about: 
```
Call to sdw_update_slave_status() needs to be added to sdw_clear_slave_status()
to ensure Slaves are informed of the new status via update_status() callback.
```

Not sure that gets the whole message, but something along those lines is easier to understand. Basicly when setting a new set, modify() needs to be paired with update(), right?

Reply:
Thanks @kv2019i for the commit message. Yes, modify() needs to be paired with update().

Reply:
if modify() needs to be paired with update(), why not call update() from within modify? Just wondering why only here and not in the other places where we call modify

Reply:
We may need to do something between modify and update according to what status it is. We use [sdw_handle_slave_status](https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/soundwire/bus.c#L1715) to handle Slave status change event and we call modify and update within `sdw_handle_slave_status`. The other case is [sdw_clear_slave_status](https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/soundwire/bus.c#L1827), we use `sdw_clear_slave_status` to force Slave reattach in resume and somehow we didn't call `sdw_update_slave_status` within `sdw_clear_slave_status`